archinstall/lib/installer.py

```diff
@@ -1070,6 +1070,26 @@ def install_profile(self, profile: Profile):
 		self.log(f'Installing archinstall profile(s) {absolute_name}', level=logging.INFO)
 		profile.install(self)
 
+	def install_audio(self, audio: str, enable_for_users: List[User] = []):
```

Comment:
I like how clean this function is, but there is a risk `Installer()` grows out of control. Which is why I would argue we keep the `/applications/<app>` handlers for specific applications.

Thoughts?

Reply:
My though on this was that the installer is a dedicated class that can handle all installation specific things by simply calling a function which defined what will be installed, e.g.
```
installer.install_profile(...)
installer.install_audio(...)
installer.install_(...)
installer.install_bootloader(...)
```

This would make documentation quite simple and in case of users wanting to script their own installation it will make things less disconnected. The installer is self-contained and no installer instance or session needs to be passed around which also helps with the dependency hell :)

I'm not too worried about the installer size tbh, I think it'll need a cleanup all together and all exposed functions that are handling specific installation steps can be nicely highlighted. But I'm happy to discuss it :)

Reply:
I don't mind splitting out single things into a separate class as well if that helps, but just looking at the pipewire setup, I'm not entirely sure I fully understand why it was actually implemented as a profile (maybe that was also me when doing the profile rewrite, it was a long time ago...)

Reply:
I don't mind if we have a wrapper in installer for those things.
But the bulk task of performing actions could still be in the application profile.

I also do outside of installations:
```
import archinstall
archinstall.Application(installation, 'postgresql').install()
```

These serve as helpers outside of installations as well.
For instance for post-installation scripts and what not.

I get that it might seem weird considering the profiles use `archinstall.storage['installation_session']` quite extensively. But that's something that can be injected at runtime with dummy instances as well as something I was hoping we could move away from some how.

Reply:
Another reason why I like dividing them up into their own little files is that it's easier to find, add and modify the specific application steps. Digging through `Installer()` is probably a bit intimidating.

Reply:
Teah I'm not a fan of throwing the session around and storing in it global places that very much defeats the purpose of isolating logic and providing a interface. 

I'm happy to revert this for now and keep it as a profile, we can look at this in a follow up

Reply:
Sounds good! Let's pool our experiences together and make a kick ass fix for the profiles after this one.

Makes this PR a bit easier to test too so thank you, you're doing awesome with this massive PR